tests/python/pants_test/backend/jvm/targets/test_jvm_binary.py

```diff
@@ -144,31 +185,43 @@ def test_jar_rules_field(self):
     self._assert_fingerprints_not_equal([field1, field2, field3, field4, field5, field6, field7])
 
   def test_manifest_entries(self):
-    target = self.make_target(':foo',
-                              JvmBinary,
-                              main='com.example.Foo',
-                              manifest_entries={'Foo-Field': 'foo'})
+    self.add_to_build_file(
+      '',
+      '''jvm_binary(
+  name = "foo",
+  main = "com.example.Foo",
+  manifest_entries = {"Foo-Field": "foo"},
+)''')
+    target = self.target(":foo")
     self.assertTrue(isinstance(target.payload.manifest_entries, ManifestEntries))
     entries = target.payload.manifest_entries.entries
     self.assertEquals({'Foo-Field': 'foo'}, entries)
 
   def test_manifest_not_dict(self):
+    self.add_to_build_file(
+      '',
+      '''jvm_binary(
+  name = "foo",
+  main = "com.example.Foo",
+  manifest_entries = "foo",
+)''')
     with self.assertRaisesRegexp(TargetDefinitionException,
                                  r'Invalid target JvmBinary.*foo.*: manifest_entries must be a '
-                                 r'dict. got unicode'):
-      self.make_target(':foo',
-                       JvmBinary,
-                       main='com.example.Foo',
-                       manifest_entries='foo')
+                                 r'dict. got (str|unicode)'):
+      self.target(':foo')
 
   def test_manifest_bad_key(self):
-    with self.assertRaisesRegexp(ManifestEntries.ExpectedDictionaryError,
-                                 r'^entries must be dictionary of strings, got key .* '
-                                 r'type JarDependency$'):
-      self.make_target(':foo',
-                       JvmBinary,
-                       main='com.example.Foo',
-                       manifest_entries={JarDependency('bad', 'bad', 'bad'): 'foo'})
+    self.add_to_build_file(
+      '',
+      '''jvm_binary(
+  name = "foo",
+  main = "com.example.Foo",
+  manifest_entries = {jar_dependency("bad", "bad", "bad"): "foo"},
+)''')
+    with self.assertRaisesRegexp(TargetDefinitionException,
```

Comment:
This is a slight change in test expectation, as we're now catching the exception a layer out, so it's wrapped up a little